### PR TITLE
[#10] Integrate survey screen

### LINKED ISSUE #10: [Integrate] As a user, I can see surveys
## Why
Implement integrations for Home screen

## Acceptance Criteria
- App must be able to show loading while fetching data
- App must be able to show surveys after fetching data
- Fetched data will be stored as cache
- The cached data will be used to display on the next time users enter the Home screen, or relaunch; in the meantime, the app should fetch data from the API as well, if there are any changes, then data should be propagated to the Home data.
- When I pull to refresh, surveys got updated and also update the cache
- The navigation indicator list (bullets) must be dynamic and based on the API response.

## Resources
[API](https://documenter.getpostman.com/view/11835486/UVsPNPfU#deb6f64d-60a4-434b-9c83-215ad8a07568)

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
Closes #10 

## What happened 👀

App must be able to show survey list. Survey list will have swipe function and pull to refresh feature. When swiped, next survey will be shown. Cache data to show next time when app is relaunched. During the time survey list will be reload and if there is an update, data will be propagated.

## Insight 📝

- Implement survey data fetching
- Implement survey data caching with room
- Implement pull to refresh feature
- When pulled to refresh all survey data will be cleared and survey list will be fetched again from page 1

## Proof Of Work 📹

https://user-images.githubusercontent.com/32578035/187044119-bb2b982d-212b-4e01-a1a4-3640a22cdfe9.mp4

### CODE REVIEW

**Comment by ryan-conway** [comment]
Shouldn't the data fetching and caching be implemented in the [backend ticket](#18 ), not the integration ticket? AFAIK backend tickets are for data/domain related tasks such as network calls, use cases, etc., while integration tickets are for hooking up our UI and our backend together 🤔 
CC: @AVI5HEK

**Comment by kaungkhantsoe** [comment]
@ryan-conway You are right. Let me update the PR

File: app/src/androidTest/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/FakeHomeRepo.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import com.kks.nimblesurveyjetpackcompose.model.ResourceState
+import com.kks.nimblesurveyjetpackcompose.model.entities.Survey
+import com.kks.nimblesurveyjetpackcompose.model.response.SurveyResponse
+import com.kks.nimblesurveyjetpackcompose.model.response.UserResponse
+import com.kks.nimblesurveyjetpackcompose.repo.home.HomeRepo
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+
+class FakeHomeRepo : HomeRepo {
```
Comment (ryan-conway) [inline app/src/androidTest/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/FakeHomeRepo.kt]: Do we really need a fake for this? Why not just use a mock in the tests instead? 🤔

File: app/src/androidTest/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreenKtTest.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import dagger.hilt.android.testing.UninstallModules
+import kotlinx.coroutines.Dispatchers
+import org.junit.Before
+import org.junit.Rule
+
+@UninstallModules(RepositoryModule::class)
+@HiltAndroidTest
+class HomeScreenKtTest : BaseAndroidComposeTest() {
```
Comment (ryan-conway) [inline app/src/androidTest/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreenKtTest.kt]: Any reason why we added this class but didn't include any tests? 🤔

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt [hunk 3]
```diff
@@ -104,98 +119,124 @@ fun HomeScreen(viewModel: HomeViewModel = hiltViewModel()) {
...
+                        survey = surveyList[selectedSurveyNumber],
+                        userAvatar = userAvatar
                     )
                 }
             }
+            error?.let { error ->
+                item {
+                    Box(
+                        modifier = Modifier
+                            .fillMaxSize(),
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt]:
```suggestion
                        modifier = Modifier.fillMaxSize(),
```

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt [hunk 4]
```diff
@@ -104,98 +119,124 @@ fun HomeScreen(viewModel: HomeViewModel = hiltViewModel()) {
...
 @Composable
 fun SurveyContent(
     modifier: Modifier,
-    surveyResponse: SurveyResponse,
+    survey: Survey,
     numberOfPage: Int,
-    selectedSurveyNumber: Int
+    selectedSurveyNumber: Int,
+    userAvatar: String?
 ) {
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt]: Modifier, by convention, is usually the first optional parameter in a composable function 👀 
```kotlin
fun SurveyContent(
    survey: Survey,
    numberOfPages: Int,
    selectedPage: Int,
    userAvatar: String?,
    modifier: Modifier = Modifier
) {
```
or since `userAvatar` is nullable:
```kotlin
fun SurveyContent(
    survey: Survey,
    numberOfPages: Int,
    selectedPage: Int,
    modifier: Modifier = Modifier,
    userAvatar: String? = null
) {

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt [hunk 5]
```diff
@@ -104,98 +119,124 @@ fun HomeScreen(viewModel: HomeViewModel = hiltViewModel()) {
...
 }
 
 @Composable
-fun BottomView(
-    modifier: Modifier,
-    surveyResponse: SurveyResponse,
-    numberOfPage: Int,
-    currentPage: Int
-) {
+fun BottomView(modifier: Modifier, survey: Survey, numberOfPage: Int, currentPage: Int) {
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt]:
```suggestion
fun BottomView(
    modifier: Modifier, 
    survey: Survey, 
    numberOfPage: Int, 
    currentPage: Int) 
{
```

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt [hunk 6]
```diff
@@ -104,98 +119,124 @@ fun HomeScreen(viewModel: HomeViewModel = hiltViewModel()) {
...
+
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt]:
```suggestion
```

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt [hunk 7]
```diff
@@ -207,28 +248,37 @@ fun BottomView(
         )
         Spacer(modifier = Modifier.height(26.dp))
         SurveyText(
-            text = surveyResponse.attributes?.title.orEmpty(),
+            text = survey.title,
             fontSize = 28.sp,
             fontWeight = FontWeight.Bold,
             modifier = Modifier.padding(start = 20.dp, end = 20.dp)
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt]: `padding()` has a `horizontal` argument 😉 
```suggestion
            modifier = Modifier.padding(horizontal = 20.dp)
```

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt [hunk 8]
```diff
@@ -207,28 +248,37 @@ fun BottomView(
...
+        ConstraintLayout(modifier = Modifier.fillMaxWidth()) {
+            val (description, detailBtn) = createRefs()
             SurveyText(
-                text = surveyResponse.attributes?.description.orEmpty(),
+                text = survey.description,
                 fontSize = 17.sp,
                 color = White70,
                 modifier = Modifier
-                    .weight(1f)
                     .padding(start = 20.dp, end = 20.dp)
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt]: Same here 👀 
```suggestion
                    .padding(horizontal = 20.dp)
```

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreenShimmerLoading.kt [hunk 9]
```diff
@@ -40,11 +40,9 @@ fun HomeScreenShimmerLoading(modifier: Modifier) {
 
 @Suppress("DestructuringDeclarationWithTooManyEntries")
 @Composable
-fun ShimmerHomeScreen(brush: Brush, modifier: Modifier) {
+fun ShimmerHomeScreen(modifier: Modifier, brush: Brush) {
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreenShimmerLoading.kt]: Same comment re optional modifier 👀 
```suggestion
fun ShimmerHomeScreen(brush: Brush, modifier: Modifier = Modifier) {
```

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/splash/SplashScreen.kt [hunk 10]
```diff
@@ -51,7 +51,7 @@ import com.ramcosta.composedestinations.navigation.EmptyDestinationsNavigator
 import kotlinx.coroutines.launch
 
 private const val SPLASH_TIME = 2000L
-private const val TWEEN_ANIM_TIME = 1000
+const val TWEEN_ANIM_TIME = 1000
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/splash/SplashScreen.kt]: Optional: Move this to a new `Constants` file 🤔

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/util/DateUtil.kt [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+import java.util.*
+
+const val EEEE_MMM_D = "EEEE, MMM d"
+
+object DateUtil {
+
+    private val EEEE_comma_MMMMM_d =
+        SimpleDateFormat(EEEE_MMM_D, Locale.ENGLISH)
+
+    fun getBeautifiedCurrentDate(): String = EEEE_comma_MMMMM_d.format(Date())
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/util/DateUtil.kt]: Let's keep things reusable 😉 
```suggestion
    fun getBeautifiedDate(date: Date = Date()): String = EEEE_comma_MMMMM_d.format(date)
```

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/home/HomeViewModel.kt [hunk 12]
```diff
@@ -1,27 +1,130 @@
...
+            totalNumberOfPage = DEFAULT_NUMBER_OF_PAGE
+            getSurveyListFromDb()
+            getSurveyList()
+        }
+    }
+
+    private fun getSurveyListFromDb() {
+        viewModelScope.launch(ioDispatcher) {
+            homeRepo.getSurveyListFromDb().collect {
+                _surveyList.value = it
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/home/HomeViewModel.kt:92]: Surely we should expose a domain-level `Survey` object, not the database entity directly? 🤔

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/splash/SplashViewModel.kt [hunk 13]
```diff
@@ -44,7 +44,7 @@ class SplashViewModel @Inject constructor(
     fun startTimerToNavigateToLogin(splashTime: Long) {
         viewModelScope.launch(ioDispatcher) {
             delay(splashTime)
-            if (preferenceManager.getBooleanData(PREF_LOGGED_IN)) {
+            if (preferenceManager.getStringData(PREF_REFRESH_TOKEN).orEmpty().isNotEmpty()) {
                 _isLoginSuccess.value = true
             } else {
                 _shouldNavigateToLogin.value = true
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/splash/SplashViewModel.kt]:
```suggestion
            if (preferenceManager.getStringData(PREF_REFRESH_TOKEN).isNullOrEmpty()) {
                _shouldNavigateToLogin.value = true
            } else {
                _isLoginSuccess.value = true
```

**Review by ryan-conway — COMMENTED** [review]
First round 🙏🏻

File: app/src/androidTest/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/FakeHomeRepo.kt (hunk 1 above)
Comment (kaungkhantsoe) [reply app/src/androidTest/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/FakeHomeRepo.kt]: In fact, it depends on the dev's preference. Some like `mock` and some favor `Test Doubles`

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/home/HomeViewModel.kt (hunk 12 above)
Comment (kaungkhantsoe) [reply app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/home/HomeViewModel.kt:92]: @ryan-conway Added and exposed `SurveyModel` 👍

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/model/SurveyModel.kt [hunk 14]
```diff
@@ -0,0 +1,8 @@
+package com.kks.nimblesurveyjetpackcompose.model
+
+data class SurveyModel(
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/model/SurveyModel.kt]: How about we rename this to `Survey` and rename `entities/Survey` to `SurveyEntity`? 🤔

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/common/DotsIndicator.kt [hunk 15]
```diff
@@ -21,13 +27,17 @@ fun DotsIndicator(
     selectedColor: Color,
     unSelectedColor: Color,
     indicatorSize: Dp,
-    space: Dp
+    space: Dp,
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/common/DotsIndicator.kt]:
```suggestion
    space: Dp
```

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/util/DateUtil.kt [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+import java.util.*
+
+const val EEEE_MMM_D = "EEEE, MMM d"
+
+object DateUtil {
+
+    private val EEEE_comma_MMMMM_d =
+        SimpleDateFormat(EEEE_MMM_D, Locale.ENGLISH)
+
+    fun getBeautifiedCurrentDate(date: Date = Date()): String = EEEE_comma_MMMMM_d.format(date)
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/util/DateUtil.kt]: 😉 
```suggestion
    fun getBeautifiedDate(date: Date = Date()): String = EEEE_comma_MMMMM_d.format(date)
```

File: app/src/androidTest/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreenTest.kt [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+        with(composeTestRule) {
+            every { homeRepo.fetchSurveyList(any(), any(), any()) } returns flowOf(ResourceState.Loading)
+            every { homeRepo.getSurveyListFromDb() } returns flowOf(emptyList())
+            setupHomeComposeRule()
+            onNodeWithContentDescription(getString(R.string.home_shimmer)).assertIsDisplayed()
+        }
+    }
+
+    @Test
+    fun when_get_survey_list_is_success_show_first_survey() {
```
Comment (AVI5HEK) [inline app/src/androidTest/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreenTest.kt]:
```suggestion
    fun when_get_survey_list_is_successful_show_first_survey() {
```

**Review by ryan-conway — COMMENTED** [review]
The rest LGTM 👍🏻

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/di/RepositoryModule.kt [hunk 18]
```diff
@@ -15,8 +18,14 @@ import dagger.hilt.android.components.ViewModelComponent
 object RepositoryModule {
     @Provides
     fun provideLoginRepo(
-        api: Api,
+        apiInterface: Api,
```
Comment (AVI5HEK) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/di/RepositoryModule.kt]: Why the change? 😄

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/di/RepositoryModule.kt [hunk 19]
```diff
@@ -15,8 +18,14 @@ import dagger.hilt.android.components.ViewModelComponent
...
         preferenceManager: PreferenceManager,
         customKeyProvider: CustomKeyProvider
-    ): LoginRepo = LoginRepoImpl(api, preferenceManager, customKeyProvider)
+    ): LoginRepo = LoginRepoImpl(apiInterface, preferenceManager, customKeyProvider)
+
+    @Provides
+    fun provideHomeRepoImpl(
+        apiInterface: Api,
```
Comment (AVI5HEK) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/di/RepositoryModule.kt]: Here too, simply `api` would be fine.

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/common/DotsIndicator.kt (hunk 15 above)
Comment (AVI5HEK) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/common/DotsIndicator.kt]: Maybe this was a mistake, but TBH, I don't mind having a trailing comma on the parameter list.

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt [hunk 20]
```diff
@@ -53,45 +67,57 @@ fun HomeScreen(viewModel: HomeViewModel = hiltViewModel()) {
...
-                        if (currentSwipeState < sizePx / 2 && threshold == IDLE) {
+                        if (currentSwipeState < endAnchor / 2 && threshold == IDLE) {
                             // Swipe to left
                             threshold = LEFT_SWIPE
-                        } else if (currentSwipeState > sizePx / 2 && threshold == IDLE) {
+                        } else if (currentSwipeState > endAnchor / 2 && threshold == IDLE) {
                             // Swipe to right
                             threshold = RIGHT_SWIPE
-                        } else if ((currentSwipeState == sizePx / 2)) {
+                        } else if ((currentSwipeState == endAnchor / 2)) {
```
Comment (AVI5HEK) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt]:
```suggestion
                        } else if (currentSwipeState == endAnchor / 2) {
```

Comment (AVI5HEK) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt]: When we need more than 2 if/else blocks, a `when` block is more readable instead, IMO.

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/util/DateUtil.kt [hunk 21]
```diff
@@ -0,0 +1,14 @@
+package com.kks.nimblesurveyjetpackcompose.util
+
+import java.text.SimpleDateFormat
+import java.util.*
+
+const val EEEE_MMM_D = "EEEE, MMM d"
+
+object DateUtil {
+
+    private val EEEE_comma_MMMMM_d =
```
Comment (AVI5HEK) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/util/DateUtil.kt]: The property name should be lower camel case by convention.

**Review by AVI5HEK — COMMENTED** [review]
First round.

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/di/RepositoryModule.kt (hunk 18 above)
Comment (kaungkhantsoe) [reply app/src/main/java/com/kks/nimblesurveyjetpackcompose/di/RepositoryModule.kt]: It changed when conflict. I am tripping since I cannot remember all the changes before 😭 . Will update it.

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt [hunk 22]
```diff
@@ -53,45 +67,61 @@ fun HomeScreen(viewModel: HomeViewModel = hiltViewModel()) {
...
         LazyColumn {
             if (isRefreshing && surveyList.isEmpty()) {
-                item { HomeScreenShimmerLoading(modifier = Modifier.fillParentMaxSize()) }
+                item {
+                    HomeScreenShimmerLoading(
+                        modifier = Modifier
+                            .fillParentMaxSize()
+                            .semantics { contentDescription = shimmerDescription }
+                    )
+                }
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt]: Let's revert this change and put the `contentDescription` directly into `ShimmerHomeScreen`
```suggestion
                item { HomeScreenShimmerLoading(modifier = Modifier.fillParentMaxSize()) }
```
And in `ShimmerHomeScreen`:
```kotlin
@Composable
fun ShimmerHomeScreen(brush: Brush, modifier: Modifier) {
    val shimmerDescription = stringResource(id = R.string.home_shimmer)
    ConstraintLayout(
        modifier = modifier
            .background(Color.Black)
            .semantics { contentDescription = shimmerDescription }
    ) {

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt [hunk 23]
```diff
@@ -53,45 +67,61 @@ fun HomeScreen(viewModel: HomeViewModel = hiltViewModel()) {
...
+    val endAnchor = LocalDensity.current.run { LocalConfiguration.current.screenWidthDp.toDp().toPx() }
     // Maps anchor points (in px) to states
-    val anchors = mapOf(0f to "L", sizePx / 2 to "M", sizePx to "R")
+    val anchors = mapOf(START_ANCHOR to LEFT_STATE, endAnchor / 2 to MID_STATE, endAnchor to RIGHT_STATE)
     var threshold by remember { mutableStateOf(IDLE) }
     val isRefreshing by viewModel.isRefreshing.collectAsState()
+    val userAvatar by viewModel.userAvatar.collectAsState()
+
+    val shimmerDescription = stringResource(id = R.string.home_shimmer)
+    val surveyContentDescription = stringResource(id = R.string.home_survey_content)
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt]: Same here

**Review by ryan-conway — COMMENTED** [review]
Last two comments 👀

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt (hunk 23 above)
Comment (ryan-conway) [reply app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt]: @kaungkhantsoe This can be moved as well. There's no point in setting the content description so far back up the hierarchy.
Similar to "state hoisting", we should only hoist details up to the composable that needs to access it and no further than that, otherwise our code becomes bloated

File: app/src/androidTest/java/com/kks/nimblesurveyjetpackcompose/AndroidTestConstants.kt [hunk 24]
```diff
@@ -0,0 +1,8 @@
+package com.kks.nimblesurveyjetpackcompose
+
+import com.kks.nimblesurveyjetpackcompose.model.Survey
+
+val surveyLists = listOf(
```
Comment (AVI5HEK) [inline app/src/androidTest/java/com/kks/nimblesurveyjetpackcompose/AndroidTestConstants.kt]: Convention.

```suggestion
val surveys = listOf(
```

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/cache/SurveyDao.kt [hunk 25]
```diff
@@ -4,16 +4,16 @@ import androidx.room.Dao
...
 import androidx.room.Query
-import com.kks.nimblesurveyjetpackcompose.model.entities.Survey
+import com.kks.nimblesurveyjetpackcompose.model.entities.SurveyEntity
 import kotlinx.coroutines.flow.Flow
 
 @Dao
 interface SurveyDao {
     @Insert(onConflict = OnConflictStrategy.REPLACE)
-    suspend fun addSurveys(surveyList: List<Survey>)
+    suspend fun addSurveys(surveyEntityList: List<SurveyEntity>)
```
Comment (AVI5HEK) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/cache/SurveyDao.kt]:
```suggestion
    suspend fun addSurveys(surveyEntities: List<SurveyEntity>)
```

**Review by AVI5HEK — COMMENTED** [review]
Some minor naming issues. The rest lgtm.

**Review by ryan-conway — APPROVED** [review]

**Review by AVI5HEK — APPROVED** [review]
LGTM 🚀
